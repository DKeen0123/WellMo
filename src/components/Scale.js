import React, { Component } from 'react';

class Scale extends Component {
  render() {
    return (
      <div>
        <h1>Scale component</h1>
      </div>
    );
  }
}

export default Scale;
