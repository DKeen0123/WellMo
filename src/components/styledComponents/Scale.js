import styled from 'styled-components';

const styledScale = styled.input`
  font-size: 22px;
  text-align: center;
  padding: 7px;
  margin-bottom: 15px;
`;

export default styledScale;
