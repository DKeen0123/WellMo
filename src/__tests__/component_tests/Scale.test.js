import React from 'react';
import { shallow, mount } from 'enzyme';
import Scale from '../../components/Scale';

describe('Scale', () => {
  let scale = shallow(<Scale />);

  it('renders correctly', () => {
    expect(scale).toMatchSnapshot();
  });

  describe('when a user types in a number', () => {
    it('fires the `handleTextBoxInput()` function', () => {
      const mockhandleTextBoxInput = jest.fn();
      scale = shallow(<Scale handleTextBoxInput={mockhandleTextBoxInput} />);
      scale.find('#firstScale').simulate('change');
      expect(mockhandleTextBoxInput).toBeCalled();
    });
  });
});
