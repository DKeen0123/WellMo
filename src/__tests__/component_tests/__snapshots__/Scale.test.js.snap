// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scale renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scale />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Scale component
</h1>,
        <input
          id="firstScale"
<<<<<<< HEAD
          max="100"
          min="0"
          name="moodScale"
          onChange={[Function]}
          type="number"
=======
          name="scaleInput"
          onChange={undefined}
          type="integer"
>>>>>>> 56fb018de3b7225f29830c1878c414b6b7572e4a
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Scale component",
        },
        "ref": null,
        "rendered": "Scale component",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "firstScale",
<<<<<<< HEAD
          "max": "100",
          "min": "0",
          "name": "moodScale",
          "onChange": [Function],
          "type": "number",
=======
          "name": "scaleInput",
          "onChange": undefined,
          "type": "integer",
>>>>>>> 56fb018de3b7225f29830c1878c414b6b7572e4a
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Scale component
</h1>,
          <input
            id="firstScale"
<<<<<<< HEAD
            max="100"
            min="0"
            name="moodScale"
            onChange={[Function]}
            type="number"
=======
            name="scaleInput"
            onChange={undefined}
            type="integer"
>>>>>>> 56fb018de3b7225f29830c1878c414b6b7572e4a
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Scale component",
          },
          "ref": null,
          "rendered": "Scale component",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "firstScale",
<<<<<<< HEAD
            "max": "100",
            "min": "0",
            "name": "moodScale",
            "onChange": [Function],
            "type": "number",
=======
            "name": "scaleInput",
            "onChange": undefined,
            "type": "integer",
>>>>>>> 56fb018de3b7225f29830c1878c414b6b7572e4a
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
